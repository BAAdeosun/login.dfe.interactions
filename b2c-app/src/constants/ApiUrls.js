export const API_URLS = {
    //TODO replace this URL with placeholder __--changeEmailAPI--__ when env vars are ready
    CHANGE_EMAIL: "https://dev-api-customerengagement.platform.education.gov.uk/authncs/ChangeEmail"
};