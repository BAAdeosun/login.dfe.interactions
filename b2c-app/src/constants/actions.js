export const ACTIONS = {
    SIGNUP: 'signup',
    LOGIN: 'login',
    RESET_PASSWORD: 'reset-password',
    FIND_EMAIL: 'find-email',
    CHANGE_EMAIL: 'change-email'
};