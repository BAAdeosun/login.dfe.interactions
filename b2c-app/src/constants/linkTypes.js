export const LINK_TYPES = {
    LINK: 'link',
    BUTTON: 'button',
    API_CALL: 'api'
};