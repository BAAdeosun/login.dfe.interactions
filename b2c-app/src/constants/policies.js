export const POLICIES = {
    SIGNIN_INVITATION: 'B2C_1A_Signin_Invitation',
    ACCOUNT_SIGNUP: 'B2C_1A_Account_Signup',
    PASSWORD_RESET: 'B2C_1A_Password_Reset',
    PASSWORD_RESET_CONFIRMATION: 'B2C_1A_Password_Reset_Confirmation',
    FIND_EMAIL: 'B2C_1A_Find_Email',
    SIGNUP_CONFIRMATION: 'B2C_1A_Signup_Confirmation',
    SIGNUP_INVITATION: 'B2C_1A_Signup_Invitation',
    CHANGE_EMAIL: 'B2C_1A_Change_Email'
};