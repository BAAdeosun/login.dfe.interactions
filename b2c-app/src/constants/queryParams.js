export const QUERY_PARAMS = {
    CLIENT_ID: 'client_id',
    REDIRECT_URI: 'redirect_uri'
}