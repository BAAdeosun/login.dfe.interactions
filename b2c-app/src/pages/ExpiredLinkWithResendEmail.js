import React from 'react';
import components from '../components';
import { ACTIONS } from '../constants/actions';

class ExpiredLinkWithResendEmail extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            showB2CErrors: true,
            errors: []
        }
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(e) {
        e.preventDefault();
        let b2cResendEmailButton = document.getElementById('continue');
        if (b2cResendEmailButton) {
            b2cResendEmailButton.click();
        }
    }

    render() {

        const pageConfig = {
            title: "Activation link expired",
            showB2CErrors: this.state.showB2CErrors,
            errors: this.state.errors
        };

        let linkExpiredText;
        let submitButtonText;

        if (this.props.action === ACTIONS.SIGNUP) {
            linkExpiredText = 'The link in your account activation email has expired.';
            submitButtonText = 'Resend activation email';
        }
        else if (this.props.action === ACTIONS.CHANGE_EMAIL) {
            linkExpiredText = 'This activation link has expired.';
            submitButtonText = 'Request new activation link';
        }

        const linkExpiredParagraph =
            <components.Paragraph>
                {linkExpiredText}
            </components.Paragraph>

        const formContent = <div></div>

        const content = linkExpiredParagraph;

        const columns = [
            {
                header: pageConfig.title,
                aboveFormContent: content,
                formContent: formContent,
                submitButtonText: submitButtonText,
                submitHandler: this.handleSubmit
            }
        ];

        return (

            <div id="expiredLink" >
                <components.PageContainer pageConfig={pageConfig} columns={columns} />
            </div>
        )
    }
}

export default ExpiredLinkWithResendEmail;
