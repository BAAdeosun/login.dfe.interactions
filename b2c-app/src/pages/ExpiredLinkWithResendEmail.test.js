
import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { ACTIONS } from '../constants/actions';
import { JSDOM } from "jsdom";

import ExpiredLinkWithResendEmail from './ExpiredLinkWithResendEmail';

it('renders without crashing', () => {
    shallow(<ExpiredLinkWithResendEmail />);
});

it('renders correctly without action passed in', () => {
    const tree = renderer
        .create(<ExpiredLinkWithResendEmail />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with action ACTIONS.SIGNUP passed in', () => {
    const tree = renderer
        .create(<ExpiredLinkWithResendEmail action={ACTIONS.SIGNUP} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with action ACTIONS.CHANGE_EMAIL passed in', () => {
    const tree = renderer
        .create(<ExpiredLinkWithResendEmail action={ACTIONS.CHANGE_EMAIL} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('when clicking the resend button (submitting custom form)', () => {

    let wrapper;
    let form;
    let b2cSubmitSpy;

    beforeEach(() => {
        //mount page with values in state for all required fields
        wrapper = mount(<ExpiredLinkWithResendEmail />);

        //get form element to simulate submit in tests
        form = wrapper.find('#customForm');

        //create custom DOM that we will use to simulate the content added by B2C
        const dom = new JSDOM();
        global.document = dom.window.document;

    });

    describe('when all B2C DOM elements are present', () => {

        it('calls the B2C submit button click', () => {

            //add the B2C submit button at least and spy on its click event
            let b2cSubmitElem = global.document.createElement('button');
            b2cSubmitElem.id = 'continue';
            global.document.body.appendChild(b2cSubmitElem);

            //spy on b2c submit button click
            b2cSubmitSpy = jest.spyOn(global.document.getElementById('continue'), 'click');

            form.simulate('submit');
            expect(b2cSubmitSpy).toHaveBeenCalled();
        });
    });



});