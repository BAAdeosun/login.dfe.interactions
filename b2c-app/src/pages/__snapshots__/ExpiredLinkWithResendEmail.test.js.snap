// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with action ACTIONS.CHANGE_EMAIL passed in 1`] = `
<div
  id="expiredLink"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <div
        className="pageLevelErrorContainer hide"
      >
        <div
          aria-labelledby="error-summary-title"
          className="govuk-error-summary"
          data-module="govuk-error-summary"
          role="alert"
          tabIndex="-1"
        >
          <h2
            className="govuk-error-summary__title"
            id="error-summary-title"
          >
            There is a problem
          </h2>
          <div
            className="govuk-error-summary__body"
          >
            <ul
              className="govuk-list govuk-error-summary__list"
              id="errorSummaryItems"
            />
          </div>
        </div>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              Activation link expired
            </h1>
            <p
              className="govuk-body"
            >
              This activation link has expired.
            </p>
            <form
              id="customForm"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div />
              <button
                className="govuk-button"
                id="preSubmit"
                type="submit"
              >
                Request new activation link
              </button>
            </form>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`renders correctly with action ACTIONS.SIGNUP passed in 1`] = `
<div
  id="expiredLink"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <div
        className="pageLevelErrorContainer hide"
      >
        <div
          aria-labelledby="error-summary-title"
          className="govuk-error-summary"
          data-module="govuk-error-summary"
          role="alert"
          tabIndex="-1"
        >
          <h2
            className="govuk-error-summary__title"
            id="error-summary-title"
          >
            There is a problem
          </h2>
          <div
            className="govuk-error-summary__body"
          >
            <ul
              className="govuk-list govuk-error-summary__list"
              id="errorSummaryItems"
            />
          </div>
        </div>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              Activation link expired
            </h1>
            <p
              className="govuk-body"
            >
              The link in your account activation email has expired.
            </p>
            <form
              id="customForm"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div />
              <button
                className="govuk-button"
                id="preSubmit"
                type="submit"
              >
                Resend activation email
              </button>
            </form>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`renders correctly without action passed in 1`] = `
<div
  id="expiredLink"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <div
        className="pageLevelErrorContainer hide"
      >
        <div
          aria-labelledby="error-summary-title"
          className="govuk-error-summary"
          data-module="govuk-error-summary"
          role="alert"
          tabIndex="-1"
        >
          <h2
            className="govuk-error-summary__title"
            id="error-summary-title"
          >
            There is a problem
          </h2>
          <div
            className="govuk-error-summary__body"
          >
            <ul
              className="govuk-list govuk-error-summary__list"
              id="errorSummaryItems"
            />
          </div>
        </div>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              Activation link expired
            </h1>
            <p
              className="govuk-body"
            />
            <form
              id="customForm"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div />
              <button
                className="govuk-button"
                id="preSubmit"
                type="submit"
              />
            </form>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
